### DISCUSSION TITLE
Problemas com login

### DISCUSSION THREAD

**Discussion by mister-vini** [frontendbr/forum#1091] [Perguntas e Respostas]
Boa tarde pessoal, estou tendo um problema ao fazer login, quando faço login para renderizar minha navbar eu preciso dar um reload, tem como eu fazer sem estar fazendo reload na pagina? e tem algo haver com o lifecycle?

**Comment by mister-vini** [comment]
![antes login](https://user-images.githubusercontent.com/38114375/40192228-27954eea-59da-11e8-8f31-e1f206856c2b.PNG)
![apos login](https://user-images.githubusercontent.com/38114375/40192229-27b53d72-59da-11e8-9c75-3764b329ff39.PNG)

**Comment by mister-vini** [comment]
Precisaria renderizar minha navbar

**Comment by torto** [comment]
Está usando qual framework? Coloque um exemplo do cód. assim podemos ajudar melhor :heart:

**Comment by mister-vini** [comment]
Estou usando ReactJS 
![image](https://user-images.githubusercontent.com/38114375/40192697-81d15b50-59db-11e8-81ef-faead55f598b.png)

**Comment by torto** [comment]
Oq vc pode fazer eh separar o render do menu com duas funções.. uma que não é logada e outra logada.. dai no render principal tu faz um if validando se existe login ou não... mais ou menos assim:

```javascript
render() {
   if(isLogged) return this.menuIsLogged()
   return this.menuNotLogged()
}

menuIsLogged() {
    return (<ul>
         <li>Home</li>
         <li>Restaurantes</li>
         <li>Login</li>
         <li>Cadastre-se</li>
      </ul>)
}

menuNotLogged() {
     return (<ul>
         <li>Home</li>
         <li>Restaurantes</li>
         <li>Sair</li>
      </ul>)
}

isLogged() {
    const { loginCredentials } = this.props
    return !!loginCreditials
}
```

**Comment by mister-vini** [comment]
Entendi obrigado @torto voce sabe me dizer se eu preciso chamar no lifecycle para renderizar?
